flink-filesystems/flink-s3-fs-presto/src/test/java/org/apache/flink/fs/s3presto/PrestoS3FileSystemTest.java

```diff
@@ -97,10 +98,15 @@ public void testShadingOfAwsCredProviderConfig() {
     //  utilities
     // ------------------------------------------------------------------------
 
-    private static void validateBasicCredentials(FileSystem fs) throws Exception {
-        assertTrue(fs instanceof FlinkS3FileSystem);
+    private static void validateBasicCredentials(FileSystem wrappedFileSystem) throws Exception {
+        assertTrue(wrappedFileSystem instanceof WrappingProxy);
```

Comment:
Why is this change needed now? Even in the previous version we used `ClassLoaderFixingFileSystem` which is where, I guess, the `WrappingProxy` is coming from. 🤔

Reply:
This FileSystem didn't come from plugin loader, so it hasn't been wrapped until this change

Reply:
Ah, good point. I forgot that we added the safe guard also to the default filesystem loading when writing this comment. 👍